src/Logging.cpp

```diff
@@ -176,6 +188,36 @@ bulk_log(string_view message)
   do_log(message, true);
 }
 
+bool
+results()
+{
+  return use_statslog;
+}
+
+void
+log_result(const std::string& input, nonstd::optional<std::string> result)
+{
+  if (result) {
+    auto path = statslog_path;
+    if (Util::ends_with(path, ".d")) {
+      Util::ensure_dir_exists(path);
+      path = FMT("{}/stats{}.log", path, getpid() % 256);
```

Comment:
That sounds OK, it might have been a case of premature optimization (or perhaps paranoia, when I added the locks)

Removing the lockfile while also removing the directory sounds safe enough. I think it was mostly "unsafe" for longer lines ?

Reply:
>  I think it was mostly "unsafe" for longer lines ?

Correct.